src/custom/hooks/useChangeNetworks.ts

```diff
@@ -1,82 +1,113 @@
-import { useCallback, useEffect, useRef } from 'react'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import { useOnClickOutside } from 'hooks/useOnClickOutside'
 import { useActiveWeb3React } from 'hooks/web3'
 import { switchToNetwork } from 'utils/switchToNetwork'
-import { useModalOpen, useToggleModal } from '../state/application/hooks'
+import { useModalOpen, useToggleModal, useWalletModalToggle } from 'state/application/hooks'
 import { CHAIN_INFO } from 'constants/chainInfo'
-import useParsedQueryString from '@src/hooks/useParsedQueryString'
-import usePrevious from '@src/hooks/usePrevious'
-import { addPopup, ApplicationModal } from '@src/state/application/reducer'
-import { useAppDispatch } from '@src/state/hooks'
-import { replaceURLParam } from '@src/utils/routes'
+import useParsedQueryString from 'hooks/useParsedQueryString'
+import usePrevious from 'hooks/usePrevious'
+import { addPopup, ApplicationModal } from 'state/application/reducer'
+import { useAppDispatch } from 'state/hooks'
+import { replaceURLParam } from 'utils/routes'
 import { getChainNameFromId, getParsedChainId } from 'components/Header/NetworkSelector'
 import { useHistory } from 'react-router-dom'
+import { supportedChainId } from 'utils/supportedChainId'
 
-type ChangeNetworksParams = Pick<ReturnType<typeof useActiveWeb3React>, 'chainId' | 'library'>
+type ChangeNetworksParams = Pick<ReturnType<typeof useActiveWeb3React>, 'account' | 'chainId' | 'library'>
+export type ChainSwitchCallbackOptions = { skipWalletToggle: boolean; skipToggle: boolean }
 
 /**
  * Hook extracted from Header/NetworkSelector component pretty much verbatim
  *
  * @param chainId
  * @param library
  */
```

Comment:
Can you update the doc string as well?

Feel free to remove the params bit, auto generated stuff

Reply:
Remove the `@param` lines, and update the comment, as it's not verbatin the same code as we had in the `Header/NetworkSelector`.
Or remove the whole docstring